### PR TITLE
[FIX] DCOS-2360 Error 'NoneType' object has no attribute 'url'

### PR DESCRIPTION
**Pull request by ytalashkocv**

### CODE REVIEW

**Comment by ytalashkocv** [comment]
@mgummelt Please review

File: dcos/http.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def _default_to_exception(response):
        not isinstance(response, requests.exceptions.RequestException):
         return response
 
-    url = response.request.url
+    url = response.request.url if response.request else 'no URL'
```
Comment (mgummelt) [inline dcos/http.py]: The bug here was that I was assuming all RequestException objects have a non-None `request` property, which is not true.

So the solution here is to use the `response.request.url` property only for those exceptions that support it, so you should remove this line, and just use `response.request.url` directly in the below references to `url`

Comment (ytalashkocv) [inline dcos/http.py]: Done

**Comment by ytalashkocv** [comment]
@mgummelt PR updated. Please review

**Comment by mgummelt** [comment]
LGTM.  Thanks!

**Pull request merged**